stock_picking_type_routing_operation/models/stock_move.py

```diff
@@ -33,36 +31,41 @@ def _split_per_zone(self):
 
             # We'll split the move to have one move per different zones where
             # we have to take products
-            zone_quantities = {}
+            routing_quantities = {}
             for source_location, qty in move_lines.items():
-                zone = pick_type_model._find_zone_for_location(source_location)
-                zone_quantities.setdefault(zone, 0.0)
-                zone_quantities[zone] += qty
+                routing_picking_type = \
+                    source_location._find_picking_type_for_routing_operation()
+                routing_quantities.setdefault(routing_picking_type, 0.0)
+                routing_quantities[routing_picking_type] += qty
 
-            if len(zone_quantities) == 1:
+            if len(routing_quantities) == 1:
                 # The whole quantity can be taken from only one zone (a
                 # non-zone being equal to a zone here), nothing to split.
                 continue
 
             move._do_unreserve()
             move_to_assign_ids.add(move.id)
-            zone_location = zone.default_location_src_id
-            for zone, qty in zone_quantities.items():
+            # FIXME routing_picking_type is the last element from loop above
+            #  not sure it's correct
+            routing_location = routing_picking_type.default_location_src_id
```

Comment:
To move inside the loop below with `routing_location = picking_type.default_location_src_id`